elasticapm/traces.py

```diff
@@ -112,53 +108,54 @@ def to_dict(self):
 
 class Span(object):
     __slots__ = (
-        "idx",
-        "transaction_id",
-        "trace_id",
+        "id",
+        "transaction",
         "name",
         "type",
         "context",
         "leaf",
+        "timestamp",
         "start_time",
         "duration",
         "parent",
         "frames",
     )
 
-    def __init__(self, idx, transaction_id, trace_id, name, span_type, start_time, context=None, leaf=False):
+    def __init__(self, transaction, name, span_type, context=None, leaf=False):
         """
         Create a new Span
 
-        :param idx: Index of this span
+        :param transaction: transaction object that this span relates to
         :param name: Generic name of the span
         :param span_type: type of the span
-        :param start_time: start time relative to the transaction
         :param context: context dictionary
-        :param leaf: is this transaction a leaf transaction?
+        :param leaf: is this span a leaf span?
         """
-        self.idx = idx
-        self.transaction_id = transaction_id
-        self.trace_id = trace_id
+        self.start_time = _time_func()
+        self.id = "%016x" % random.getrandbits(64)
+        self.transaction = transaction
```

Comment:
any perf worries about referencing the transaction rather than pulling the bits needed out from it here in init?

Reply:
other than the transaction not being able to be GC'd before the last span is GC'd (which is anyway the normal course of execution), not really

In fact, it should use a bit less memory, less attributes per span. But that's probably not of any measurable relevance.